addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -203,6 +203,30 @@ def action_assign(self):
         self.ensure_one()
         self.picking_ids.action_assign()
 
+    def action_put_in_pack(self):
+        """ Action to put move lines with 'Done' quantities into a new pack
+        This method follows same logic to stock.picking.
+        """
+        self.ensure_one()
+        if self.state not in ('done', 'cancel'):
+            picking_move_lines = self.move_line_ids
+
+            move_line_ids = picking_move_lines.filtered(lambda ml:
+                float_compare(ml.qty_done, 0.0, precision_rounding=ml.product_uom_id.rounding) > 0
+                and not ml.result_package_id
+            )
+            if not move_line_ids:
+                move_line_ids = picking_move_lines.filtered(lambda ml: float_compare(ml.product_uom_qty, 0.0,
+                                     precision_rounding=ml.product_uom_id.rounding) > 0 and float_compare(ml.qty_done, 0.0,
+                                     precision_rounding=ml.product_uom_id.rounding) == 0)
+            if move_line_ids:
+                res = self.picking_ids[0]._pre_put_in_pack_hook(move_line_ids)
```

Comment:
self.picking_ids[0] is probably not enough.
- Ok for check destination and stock.package.destination
- Issue with carrier. It would be great to do a module delivery_stock_picking_batch since people could do batch picking using different carrier (that use different package type) and it would be problematic.

@oma-odoo

Reply:
But an addons for a usererror is too much